### ISSUE TITLE
Build Failure on LibreSSL -- no e_os2.h

### ISSUE THREAD

**Issue by lluixhi** [pyca/cryptography#2858]
Because e_os2.h is an internal header as of LibreSSL 2.1.4, it is not installed and the cffi bindings fail to build.

Relevant code in build_openssl.py:
``` python
_PRE_INCLUDE = """
#include <openssl/e_os2.h>
#if defined(OPENSSL_SYS_WINDOWS)
#include <windows.h>
#endif
"""
```

We could fix the build error by instead doing something like the following:

``` python
if sys.platform == "win32":
    _PRE_INCLUDE = """
    #include <windows.h>
    """
else:
    _PRE_INCLUDE = """"""
```

or maybe:

``` python
_PRE_INCLUDE = """
#if defined(_WIN32)
#include <windows.h>
#endif
"""
```

**Comment by reaperhulk** [comment]
`e_os2.h` is available in LibreSSL 2.2.x, but appears to have been removed again in 2.3. What wonderful compatibility they continue to provide...

Several versions of OpenSSL don't like it if you fail to include `e_os2.h` so we're probably better served pulling it in as a conditional checking for #defined LIBRESSL_VERSION_NUMBER (which of course means we'll have to pull in `opensslv.h` first). I'll look at a PR for this tomorrow.

**Comment by lluixhi** [comment]
I didn't realize that they made the header visible in 2.2.x only to hide it again in 2.3.x, sorry.

The patch works for me, thanks. :)
